homeassistant/components/media_player/lg_soundbar.py

```diff
@@ -0,0 +1,196 @@
+"""
+Support for LG soundbars.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/media_player.lg_soundbar/
+"""
+import logging
+
+from homeassistant.components.media_player import (
+    SUPPORT_SELECT_SOURCE, SUPPORT_VOLUME_MUTE, SUPPORT_VOLUME_SET,
+    SUPPORT_SELECT_SOUND_MODE, MediaPlayerDevice)
+
+from homeassistant.const import STATE_ON
+
+REQUIREMENTS = ['temescal==0.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_LG = SUPPORT_VOLUME_SET | SUPPORT_VOLUME_MUTE | SUPPORT_SELECT_SOURCE \
+             | SUPPORT_SELECT_SOUND_MODE
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the LG platform."""
+    if discovery_info is not None:
+        add_entities([LGDevice(discovery_info)], True)
+
+
+class LGDevice(MediaPlayerDevice):
+    """Representation of an LG soundbar device."""
+
+    def __init__(self, discovery_info):
+        """Initialize the LG speakers."""
+        import temescal
+
+        host = discovery_info.get('host')
+        port = discovery_info.get('port')
+
+        self._name = ""
+        self._volume = 0
+        self._volume_min = 0
+        self._volume_max = 0
+        self._function = -1
+        self._functions = []
+        self._equaliser = -1
+        self._equalisers = []
+        self._mute = 0
+        self._rear_volume = 0
+        self._rear_volume_min = 0
+        self._rear_volume_max = 0
+        self._woofer_volume = 0
+        self._woofer_volume_min = 0
+        self._woofer_volume_max = 0
+        self._bass = 0
+        self._treble = 0
+
+        self._device = temescal.temescal(host, port=port,
+                                         callback=self.handle_event)
```

Comment:
Callback registration should preferably be done in `async_added_to_hass`. We risk an error otherwise if the callback is called before the entity has been added to home assistant.